proxy/grpcproxy/cache/store.go

```diff
@@ -73,7 +77,11 @@ func (c *cache) Add(req *pb.RangeRequest, resp *pb.RangeResponse) {
 	defer c.mu.Unlock()
 
 	if req.Revision > c.compactedRev {
-		c.lru.Add(key, resp)
+		if req.Revision == 0 {
```

Comment:
if req.rev ==0, we can actually add two entries. one for req.Revision = 0 and the other for req.Revision=resp.Revision.